File: Find the Index of the First Occurrence in a String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) > len(haystack):
            return -1
        out,out_len,n,m = -1,0,len(needle),len(haystack)
        i,j=0,0
        while(i < n and j < m):

            if needle[i] != haystack[j]:
                out,out_len = -1,0
                j = 1+j-i
                i=0
            else:
                if out == -1:
                    out = j
                out_len+=1
                j+=1
                i+=1
        if out_len !=n:
            return -1
        return out

File: Find the Index of the First Occurrence in a String.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int index_out , i = 0;
        int j = 0;
        while ( i < haystack.length()){
            if(haystack[i] == needle[j]){
                if (j == needle.length()-1){
                    return i-j;
                }
                j++;
            }
            else{
                i = i-j;
                j = 0;
            }
            i++;
        }
        return -1;
        
    }
};
